apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/dashboard/connect/_components/ConnectEvaluationModal/index.tsx

```diff
@@ -0,0 +1,191 @@
+import useEvaluations from '$/stores/evaluations'
+import useEvaluationTemplates from '$/stores/evaluationTemplates'
+import Link from 'next/link'
+
+import EvaluationEditor from './Editor'
+import EvaluationList from './List'
+
+type SelectableItem = {
+  uuid: string
+  name: string
+  description: string
+  type: 'evaluation' | 'template'
+  data: EvaluationDto | EvaluationTemplateCategory
+}
+
+export default function ConnectionEvaluationModal({
+  projectId,
+  commitUuid,
+  documentUuid,
+}: {
+  projectId: string
+  commitUuid: string
+  documentUuid: string
+}) {
+  const [selectedItems, setSelectedItems] = useState<string[]>([])
+  const { execute, isPending: isConnecting } = useLatitudeAction(
+    connectEvaluationsAction,
+  )
+  const navigate = useNavigate()
+  const { data: usedEvaluations } = useEvaluations({
+    params: { documentUuid },
+  })
+  const {
+    data: evaluations,
+    isLoading: isLoadingEvaluations,
+    mutate,
+  } = useEvaluations()
+  const { data: templates, isLoading: isLoadingTemplates } =
+    useEvaluationTemplates()
+  const [searchTerm, setSearchTerm] = useState('')
+
+  const selectableItems: SelectableItem[] = useMemo(() => {
+    const evaluationItems = xorBy(
+      evaluations,
+      usedEvaluations,
+      (ev) => ev.id,
+    ).map((e) => ({
+      uuid: e.uuid,
+      name: e.name,
+      description: e.description,
+      type: 'evaluation' as const,
+      data: e,
+    }))
+    const templateItems = templates.map((t) => ({
+      uuid: t.id.toString(),
+      name: t.name,
+      description: t.description,
+      type: 'template' as const,
+      data: t,
+    }))
+    return [...evaluationItems, ...templateItems]
+  }, [evaluations, usedEvaluations, templates])
+
+  const filteredItems = useMemo(() => {
+    return selectableItems.filter((item) =>
+      item.name.toLowerCase().includes(searchTerm.toLowerCase()),
+    )
+  }, [selectableItems, searchTerm])
+
+  const handleSelectItem = (uuid: string) => {
+    setSelectedItems(
+      selectedItems.includes(uuid)
+        ? selectedItems.filter((id) => id !== uuid)
+        : [...selectedItems, uuid],
+    )
+  }
+
+  const createConnection = useCallback(async () => {
+    const templateIds = selectedItems
+      .filter((id) =>
+        templates?.some((template) => template.id.toString() === id),
+      )
+      .map(Number)
+    const evaluationUuids = selectedItems.filter((id) =>
+      evaluations?.some((evaluation) => evaluation.uuid === id),
+    )
+
+    const [data] = await execute({
+      projectId,
+      templateIds,
+      evaluationUuids,
+      documentUuid,
+    })
+
+    if (data) {
+      mutate()
+      const connectedEvaluation = data[0]!
+      navigate.push(
+        ROUTES.projects
+          .detail({ id: Number(projectId) })
+          .commits.detail({ uuid: commitUuid })
+          .documents.detail({ uuid: documentUuid })
+          .evaluations.detail(connectedEvaluation.evaluationId).root,
```

Comment:
All this connect action / service returns `evaluationId`. Not `evaluationUuid`. So I prefered not to change it at all and use id in the route (finally sorry)

Reply:
but... we agreed on using the uuid we can very easily return the evaluation uuid instead if that's the issue